### PR TITLE
upcoming: [M3-7695] - Add new `path_regex` match type to ACLB

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

Adds new `path_regex` Rule match type to ACLB. This will allow users to create rules that match on a path using regex.

## Preview 📷

| Before  | After   |
| ------- | ------- |
| Screenshot 2024-01-30 at 12 48 09 PM | Screenshot 2024-01-30 at 12 47 53 PM  |

### Prerequisites
- Check out this PR
- Switch your Cloud Manager to use the dev environemnt 
- Login to `dev-test-aglb` account for testing

### Verification steps 
- On the `aglbsqaMDDA013007` ACLB try to create/edit a rule and choose `Path Regex` as the match type
- Verify the copy for tooltips are updated 

## As an Author I have considered 🤔

- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [x] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [x] 🔐 Removing all sensitive information from the code and PR description
- [x] 🚩 Using a feature flag to protect the release
- [x] 👣 Providing comprehensive reproduction steps
- [x] 📑 Providing or updating our documentation
- [x] 🕛 Scheduling a pair reviewing session
- [x] 📱 Providing mobile support
- [x] ♿  Providing accessibility support

### CODE REVIEW

**Review by mjac0bs — APPROVED** [review]
UI changes look good -- confirmed that the new Path Regex match type is there, Path has been updated to Path Prefix, the tooltips are updated, and tests pass.

More of an API question, but should we be able to actually submit the Path Regex rule? I copied the placeholder text as regex and ended up with `Please pass a valid match field` as an error response.

https://github.com/linode/manager/assets/114685994/1c7ffa65-37ee-44cb-9ed6-a9459ac88faf

**Comment by carrillo-erik** [comment]
Same observation as Mariah when testing the `Path Regex` as the type. Also, should the input field show a red outline in addition to the error Notice?

Screenshot 2024-01-30 at 4 00 16 PM

**Comment by bnussman-akamai** [comment]
Good catch! @mjac0bs  I put this up without testing the API. I believe their work for this is still in progress. We can safely test the Cloud Manager pieces and assume the API team will merge their work soon!

> should the input field show a red outline in addition to the error Notice

This is a great point @carrillo-erik , but that is an issue on the API side. The current response is:

```json
{
  "reason": "Please pass a valid match field"
}
```

but it really should be 

```json
{
  "field" : "rules[0].match_condition.match_field",
  "reason": "Please pass a valid match field"
}
```

**Review by abailly-akamai — APPROVED** [review]

**Review by carrillo-erik — APPROVED** [review]

**Comment by bnussman-akamai** [comment]
Going to push a few copy changes required by UX

File: packages/manager/src/features/LoadBalancers/LoadBalancerDetail/Routes/utils.ts [hunk 1]
```diff
@@ -15,17 +15,19 @@ type CustomerFacingMatchFieldOption = Exclude<MatchField, 'always_match'>;
...
+  path_regex: 'Path Regex',
   query: 'Query String',
 };
 
 export const matchValuePlaceholder: Record<
   CustomerFacingMatchFieldOption,
   string
 > = {
-  header: 'x-my-header=this',
+  header: 'x-my-header:this',
```
Comment (bnussman-akamai) [inline packages/manager/src/features/LoadBalancers/LoadBalancerDetail/Routes/utils.ts:27]: This is an unrelated change that UX wanted.
